### ISSUE TITLE
DataParallel Usage

### ISSUE THREAD

**Issue by dbonafilia** [vikasverma1077/manifold_mixup#2]
When I try to use this the supervised PreActResnets with Manifold Mixup with torch.nn.DataParallel, it only returns the data from one of my GPUS. Is there a known lack of integration with the DataParallel module or am I likely doing something wrong?

**Comment by vikasverma1077** [comment]
Yes, that is a known issue. If I remember correctly, the reason was the way we have implemented Manifold Mixup currently. Currently, the forward pass of the model return the mixed-up labels,  from this function :https://github.com/vikasverma1077/manifold_mixup/blob/master/supervised/utils.py#L40

The approach to correct this problem would be to return shuffled targets : target_a and target b in the forward pass of the model and do the target mixing outside the model forward pass.
